### ISSUE TITLE
Charm in error: transaction already started

### ISSUE THREAD

**Issue by juditnovak** [canonical/tls-certificates-interface#235]
### Bug Description

When multiple certificates are revoked, the following error occurs.

```
unit-opensearch-2: 12:24:46 ERROR juju.worker.uniter.context cannot apply changes: removing secrets: transaction already started
transaction already started
unit-opensearch-2: 12:24:46 ERROR juju.worker.uniter.operation hook "certificates-relation-changed" (via hook dispatching script: dispatch) failed: removing secrets: transaction already started
```

Debugging suggests that the source is the `remove_all_revisions()` call, which is known to possibly trigger an error on potential race conditions (secret already deleted, etc.)

The issue is that: the charm goes into an `error` state shortly and then recovers. This should be avoided.

### To Reproduce

We observe the issue in complex charm workflows.

To reproduce I'd suggest to "hammer" on the `self-signed-certificats` charm with cerificate revocation requests.

### Environment

We are using a Juju 34 or Juju 3.5 environment.

Certificates libs version 3.17

### Relevant log output

```shell
See above
```

**Comment by juditnovak** [comment]
Cc: @reneradoi

**Comment by gruyaume** [comment]
Thanks for reporting this issue, we'll look into it shortly.

**Comment by kayra1** [comment]
Hello, since version 3.17 we've released both 3.23, and 4.1. I could not recreate this issue in either of the versions. There was a similar problem related to secret revisions that was already fixed in version 3.21. 

Outside of that I attempted to do a hundred revocations that did not cause an issue in any of my attempts. Are you still receiving this error within your charms?

**Comment by kayra1** [comment]
I am closing this issue since there seems to be no movement. Let us know if you ever come across it, and we will fix it.
